apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/DatasetsResourceTest.java

```diff
@@ -3756,20 +3766,27 @@ void findWithImageTruncation(JsonNode original, JsonNode expected, boolean trunc
             putAndAssert(datasetItemBatchWithImage, workspaceName, apiKey);
 
             // Creating 5 different experiment ids
-            var expectedDatasetItems = datasetItemBatchWithImage.items()
-                    .stream()
-                    .map(item -> item.toBuilder().data(ImmutableMap.of("image", expected)).build())
-                    .toList().reversed();
             var experimentIds = IntStream.range(0, 5).mapToObj(__ -> GENERATOR.generate()).toList();
-            var experimentItemsBatch = factory.manufacturePojo(ExperimentItemsBatch.class);
-            var experimentItemsBatchWithIds = experimentItemsBatch.toBuilder().experimentItems(
-                    IntStream.range(0, 5)
-                            .mapToObj(i -> new ArrayList<>(experimentItemsBatch.experimentItems()).get(i).toBuilder()
-                                    .experimentId(experimentIds.get(i))
-                                    .datasetItemId(datasetItemBatchWithImage.items().get(i).id()).build())
-                            .collect(Collectors.toUnmodifiableSet())).build();
+            List<ExperimentItem> experimentItems = IntStream.range(0, 5).mapToObj(i ->
+                    factory.manufacturePojo(ExperimentItem.class).toBuilder()
+                            .experimentId(experimentIds.get(i))
+                            .traceId(traces.get(i).id())
+                            .datasetItemId(datasetItemBatchWithImage.items().get(i).id()).build()).toList();
+                    PodamFactoryUtils.manufacturePojoList(factory, ExperimentItem.class);
+            var experimentItemsBatch = ExperimentItemsBatch.builder()
+                    .experimentItems(new HashSet<>(experimentItems)).build();
```

Comment:
Minor: `Set.of` usually cleaner (although immutable).

Reply:
Done in https://github.com/comet-ml/opik/pull/664/commits/17de2825fe6b9125c5f1563db0316fc544bb45f2